x-pack/plugins/enterprise_search/public/applications/workplace_search/views/groups/components/group_overview.tsx

```diff
@@ -0,0 +1,274 @@
+    <EuiButton color="primary" onClick={showSharedSourcesModal}>
+      {MANAGE_SOURCES_BUTTON_TEXT}
+    </EuiButton>
+  );
+  const manageUsersButton = !isFederatedAuth && (
+    <EuiButton color="primary" onClick={showManageUsersModal}>
+      {MANAGE_USERS_BUTTON_TEXT}
+    </EuiButton>
+  );
+  const sourcesTable = <SourcesTable sources={contentSources} />;
+
+  const sourcesSection = (
+    <ContentSection
+      title="Group content sources"
+      description={hasContentSources ? GROUP_SOURCES_DESCRIPTION : EMPTY_SOURCES_DESCRIPTION}
+      action={manageSourcesButton}
+    >
+      {hasContentSources && sourcesTable}
+    </ContentSection>
+  );
+
+  const usersSection = !isFederatedAuth && (
+    <ContentSection
+      title="Group users"
+      description={hasUsers ? GROUP_USERS_DESCRIPTION : EMPTY_USERS_DESCRIPTION}
+      action={manageUsersButton}
+    >
+      {hasUsers && <GroupUsersTable />}
+    </ContentSection>
+  );
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    updateGroupName();
+  };
+
+  const nameSection = (
+    <ContentSection title={NAME_SECTION_TITLE} description={NAME_SECTION_DESCRIPTION}>
+      <form onSubmit={handleSubmit}>
+        <EuiFormRow isInvalid={false}>
+          <EuiFlexGroup>
+            <EuiFlexItem grow={false}>
+              <EuiFieldText
+                isInvalid={false}
+                value={groupNameInputValue}
+                onChange={(e) => onGroupNameInputChange(e.target.value)}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButton disabled={!groupNameInputValue} onClick={updateGroupName}>
+                {SAVE_NAME_BUTTON_TEXT}
+              </EuiButton>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+      </form>
+    </ContentSection>
+  );
+
+  const deleteSection = (
+    <>
+      <EuiSpacer size="xs" />
+      <EuiHorizontalRule />
+      <EuiSpacer size="xxl" />
+      <ContentSection title={REMOVE_SECTION_TITLE} description={REMOVE_SECTION_DESCRIPTION}>
+        {confirmDeleteModalVisible && (
+          <EuiOverlayMask>
+            <EuiConfirmModal
+              onCancel={hideConfirmDeleteModal}
+              onConfirm={deleteGroup}
+              confirmButtonText={CONFIRM_REMOVE_BUTTON_TEXT}
+              title={CONFIRM_TITLE_TEXT}
+              cancelButtonText={CANCEL_REMOVE_BUTTON_TEXT}
+              defaultFocusedButton="confirm"
+            >
+              {CONFIRM_REMOVE_DESCRIPTION}
+            </EuiConfirmModal>
+          </EuiOverlayMask>
+        )}
+        <EuiButton
+          color="danger"
+          data-test-subj="DeleteGroup"
+          fill
+          onClick={showConfirmDeleteModal}
+        >
+          {REMOVE_BUTTON_TEXT}
+        </EuiButton>
+      </ContentSection>
+    </>
+  );
+
+  return (
+    <>
+      <SetPageChrome text="Group Overview" />
+      <SendTelemetry action="viewed" metric="group overview" />
```

Comment:
Just want to mention this rq before I forget:

- Metrics should be snaked_cased, they're used as telemetry obj keys which seem to be snaked cased universally, so `group_overview` here
- This isn't actually currently doing anything or incrementing any stats in reality, which is fine if you'd rather wire that up later. You'll need to update your [server telemetry collector](https://github.com/elastic/kibana/blob/master/x-pack/plugins/enterprise_search/server/collectors/workplace_search/telemetry.ts) for it to actually start recording/showing up in [the stats endpoint](http://localhost:5601/api/stats?extended=true), as well as updating [the telemetry plugin's schema JSON](https://github.com/elastic/kibana/blob/master/x-pack/plugins/telemetry_collection_xpack/schema/xpack_plugins.json#L198).